core/src/main/java/org/eclipse/hono/util/MessageHelper.java

```diff
@@ -1018,6 +1013,7 @@ public static Message newMessage(
             final boolean addDefaults,
             final boolean addJmsVendorProps) {
 
+        Objects.requireNonNull(target);
```

Comment:
Added `null` check:
If `target` were `null`, there would be an `IllegalArgumentException` thrown by the `addProperties` method here (already in the code before this PR).